internal/pkg/git/git.go

```diff
@@ -0,0 +1,49 @@
+package git
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/filesystem"
+	"github.com/keboola/keboola-as-code/internal/pkg/filesystem/aferofs"
+	"github.com/keboola/keboola-as-code/internal/pkg/log"
+)
+
+func Available() bool {
+	_, err := exec.LookPath("git")
+	return err == nil
+}
+
+func CheckoutTemplateRepository(url string, ref string, logger log.Logger) (filesystem.Fs, error) {
+	if !Available() {
+		return nil, fmt.Errorf("git command is not available, if you want to use templates from a git repository you have to install it first")
+	}
+
+	dir, err := ioutil.TempDir("", "keboola-as-code-templates-")
+	if err != nil {
+		return nil, err
+	}
+
+	cmd := exec.Command("git", "clone", "--no-checkout", url, dir)
+	cmd.Stdout = logger.DebugWriter()
+	cmd.Stderr = logger.DebugWriter()
+	cmd.Env = os.Environ()
+	cmd.Env = append(cmd.Env, "GIT_TERMINAL_PROMPT=0")
+	err = cmd.Run()
+	if err != nil {
+		return nil, fmt.Errorf(`git repository not found on url "%s"`, url)
```

Comment:
Tady obdobně jako dole. Když se pokusím udělat clone neexistující repository na githubu i na úplně neexistující tak to vrátí chybu s hláškou `exit code 128`.

Reply:
Ono by bolo asi najlepsie zachytit `stderr` a vratit ju ako chybu.
Napr. takto: https://stackoverflow.com/a/65573342

(`cmd.Run` podla vsetkeho vracia iba chybu s `exit code`, to som samozrejme z hlavy nevedel povedat)

Ak to chces aj ako debug, a vratit iba v pripade `err`, tak:
```
cmd.Stderr = io.MultiWriter(logger.DebugWriter(), stderrBuffer)
```

Moze to byt private repo, a user nema nastavene SSH kluce, napr....

Necham to na teba.
Napr. mozes osetrit `exit code 128` podmienkou (lepsie ak by si porovnaval cislo a nie string, vytiahni to z `cmd`).
A vratit nejaku custom spravu `repository not found ...`.

A pre ostatne chyby vratit `stderr`.

Reply:
Mno, tak co myslíš? https://github.com/keboola/keboola-as-code/pull/512/commits/439701970b0a20cbfc0a099165a35b531046b5e5

Reply:
Akorát teda v tom stderr může být potenciálně trochu bordel, protože např. zrovna `git clone` tam háže info o tom, že klonuje. 🤷‍♂️😅
![CleanShot 2022-02-02 at 16 42 08](https://user-images.githubusercontent.com/215660/152187024-1057f216-4f09-478b-8755-0d47b7ffe585.jpg)

Reply:
tvl. to je zjevně by design 🙈, zkusím ještě přidat ten `-q` 
![CleanShot 2022-02-02 at 16 47 06](https://user-images.githubusercontent.com/215660/152187767-b7aa4f1c-596f-44c0-82c7-98839db30b88.jpg)

Reply:
jo, dobrý, vyřešeno 😃 
![CleanShot 2022-02-02 at 16 51 45](https://user-images.githubusercontent.com/215660/152188701-3871545a-357d-4469-a258-d47b8a48a7d0.jpg)

Reply:
jj, progress casto prikazy loguju na STDERR, ... takze uz vsetko ok?

Reply:
Jj. Už to funguje očekávatelně. 🙂